### PR TITLE
provider/aws: Autoscaling Group Scheduled Actions

### PR DESCRIPTION
**Pull request by stack72**
As requested in #2367 & #4211

* [x] Schema
* [x] CRUD
* [x] Acceptance Tests
* [x] Documentation

Test Results:
```
TF_LOG=1 make testacc TEST=./builtin/providers/aws TESTARGS='-run=AWSAutoscalingSchedule' 2>~/tf.log
go generate ./...
TF_ACC=1 go test ./builtin/providers/aws -v -run=AWSAutoscalingSchedule -timeout 90m
=== RUN   TestAccAWSAutoscalingSchedule_basic
--- PASS: TestAccAWSAutoscalingSchedule_basic (163.86s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	163.882s
```

### CODE REVIEW

File: builtin/providers/aws/resource_aws_autoscaling_schedule.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+				Required: true,
+				ForceNew: true,
+			},
+			"start_time": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validateASGScheduleTimestamp,
+			},
+			"end_time": &schema.Schema{
```
Comment (jen20) [inline builtin/providers/aws/resource_aws_autoscaling_schedule.go]: We could make use of the (new) resource-level [ValidateFunc](https://github.com/hashicorp/terraform/blob/master/helper/schema/schema.go#L141) to ensure that `end_time` > `start_time` here.

File: builtin/providers/aws/resource_aws_autoscaling_schedule.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+const AWSAutoscalingScheduleTimeLayout = "2006-01-02T15:04:05Z"
```
Comment (jen20) [inline builtin/providers/aws/resource_aws_autoscaling_schedule.go]: There's no need to export this - it should start with a lower case character.

**Comment by jen20** [comment]
LGTM! I've put a couple of minor comments inline - not sure if the API validates that `end_time` > `start_time` but if so we can make use of the new resource level validation that @phinze added yesterday for that.

File: builtin/providers/aws/resource_aws_autoscaling_schedule.go (hunk 1 above)
Comment (jen20) [inline builtin/providers/aws/resource_aws_autoscaling_schedule.go]: Actually this hasn't yet been merged, sorry!

**Comment by scalp42** [comment]
@jen20 do you know if you define a single schedule resource but assign multiple ASGs (pretty much takes an array for `autoscaling_group_name `

**Comment by stack72** [comment]
@scalp42 right now, the api only allows a single autoscaling_group_name: http://docs.aws.amazon.com/cli/latest/reference/autoscaling/put-scheduled-update-group-action.html

**Comment by scalp42** [comment]
:+1:

**Comment by stack72** [comment]
@scalp42 FWIW its a great suggestion :)

**Pull request merged**